xml/smb_native_os.xml

```diff
@@ -384,6 +384,33 @@
       <param pos="2" name="os.build"/>
    </fingerprint>
 
+      <!-- Mac OS X -->
+   <fingerprint pattern="^Samba 3.0.28a-apple">
+      <description>OS X 10.6</description>
+      <example>Samba 3.0.28a-apple</example>
+      <param pos="0" name="os.vendor" value="Apple"/>
+      <param pos="0" name="os.family" value="Mac OS X"/>
+      <param pos="0" name="os.product" value="Mac OS X"/>
+      <param pos="0" name="os.device" value="General"/>
+      <param pos="0" name="os.version" value="10.6"/>
+      <param pos="0" name="service.vendor" value="Samba"/>
+      <param pos="0" name="service.product" value="Samba"/>
+      <param pos="0" name="service.version" value="3.0.28a-apple"/>
+   </fingerprint>
+
+   <fingerprint pattern="^Samba 3.0.25b-apple">
```

Comment:
Escape those periods and add a trailing anchor.  Also, even though you are looking for an explicit version, IMO it is simpler, cleaner and easier to add more in the future if you capture the version up here and use it in the `param` down below with a `pos` of 1.